include/filewriter/sharedqueue.hpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file sharedqueue.hpp
+ * @author Jeffin Sam
+ */
+#ifndef MLBOARD_SHARED_QUEUE_HPP
+#define MLBOARD_SHARED_QUEUE_HPP
+
+#include <mlboard/core.hpp>
+
+namespace mlboard {
+
+/**
+ * Class responsible to share a queue among main thread and event logger
+ * thread.
+ * 
+ * @tparam Datatype datatype of the elements queue would be holding. 
+ */
+template <typename Datatype>
+class SharedQueue
+{
+ public:
+  /**
+   * function to pop an element from the queue.
+   */ 
+  Datatype pop();
+
+  /**
+  * function to push an element in the queue.
```

Comment:
I think the first letter of the comment should be capital

Reply:
Done